platform/linux-generic/pktio/loop.c

```diff
@@ -74,24 +83,54 @@ static int loopback_open(odp_pktio_t id, pktio_entry_t *pktio_entry,
 		return -1;
 	}
 
-	snprintf(loopq_name, sizeof(loopq_name), "%" PRIu64 "-pktio_loopq",
-		 odp_pktio_to_u64(id));
 	pkt_loop->idx = idx;
 	pkt_loop->mtu = LOOP_MTU_MAX;
-	pkt_loop->loopq = odp_queue_create(loopq_name, NULL);
-
-	if (pkt_loop->loopq == ODP_QUEUE_INVALID)
-		return -1;
+	pkt_loop->loopq = ODP_QUEUE_INVALID;
 
 	loopback_stats_reset(pktio_entry);
 	loopback_init_capability(pktio_entry);
 
 	return 0;
 }
 
+static int loopback_pktout_queue_config(pktio_entry_t *pktio_entry,
+					const odp_pktout_queue_param_t *param)
+{
+	pkt_loop_t *pkt_loop = pkt_priv(pktio_entry);
+	odp_queue_param_t queue_param;
+	char queue_name[ODP_QUEUE_NAME_LEN];
+
+	/* Destroy old queue */
+	if (pkt_loop->loopq != ODP_QUEUE_INVALID) {
+		if (odp_queue_destroy(pkt_loop->loopq)) {
+			ODP_ERR("Destroying old loopback pktio queue failed\n");
+			return -1;
+		}
+	}
+
+	odp_queue_param_init(&queue_param);
+	queue_param.size = param->queue_size[0];
```

Comment:
It would be good to check param->queue_size[] values from application against min/max capas